### PR TITLE
(3) standardize ReportDrawer & AddEditEntityModal props/behavior

### PR DESCRIPTION
**Pull request by braxex**
## Description

Standardizes ReportDrawer & AddEditEntityModal props and behavior, especially hydration.

Changes:
- ReportDrawer now receives `selectedEntity`, just like AddEditEntityModal and DeleteEntityModal, and uses it to hydrate instead of separately fetching reportData.
- ReportDrawer now receives `verbiage` prop instead of several verbiage-related props (drawerTitle, drawerInfo, drawerDetails). This extends down to the underlying Drawer as well.
- AddEditEntityModal now receives `form` instead of `modalForm` prop, just like ReportDrawer.

### How to test

Test drawer behavior, including hydration, on DrawerReportPage and ModalDrawerReportPage.

### Changed Dependencies

n/a

## Code author checklist
- [x] I have performed a self-review of my code
- [x] I have added [thorough](https://bit.ly/3zPrxuZ) tests
- [x] ~~I have added analytics, if necessary~~
- [x] ~~I have updated the documentation, if necessary~~

## Reviewer checklist (two different people)
- [ ] I have done the deep review and verified the items in the above checklist are g2g
- [ ] I have done the lgtm review

### CODE REVIEW

**Comment by braxex** [comment]
Note: As part of this I also noticed we never actually wrote good tests for ReportDrawer, so we'll be making a ticket for that so we can get it done pre-deploy. 👍

File: services/ui-src/src/components/drawers/ReportDrawer.tsx [hunk 1]
```diff
@@ -1,55 +1,41 @@
...
+        formData={selectedEntity}
       />
       <Box sx={sx.footerBox}>
         <Flex sx={sx.buttonFlex}>
           {!isAdminTypeUser && (
-            <Button
-              variant="outline"
-              onClick={drawerDisclosure.onClose as MouseEventHandler}
-            >
+            <Button variant="outline" onClick={() => drawerDisclosure.onClose}>
```
Comment (ntsummers1) [inline services/ui-src/src/components/drawers/ReportDrawer.tsx:38]: Looks like this breaks the Cancel button on my end:

https://user-images.githubusercontent.com/19439679/195607157-46c08b34-239a-4d5c-8db9-48dad30b7283.mov

Comment (braxex) [reply services/ui-src/src/components/drawers/ReportDrawer.tsx:38]: huh, weird. totally my bad. i'll switch it back. thank you!

**Review by ntsummers1 — APPROVED** [review]

**Review by karla-vm — APPROVED** [review]
Working for me locally 👍🏽

**Pull request merged**